mono/mini/mini-x86.c

```diff
@@ -4203,8 +4203,8 @@ mono_arch_output_basic_block (MonoCompile *cfg, MonoBasicBlock *bb)
 			int nursery_shift, card_table_shift;
 			gpointer card_table_mask;
 			size_t nursery_size;
-			gulong card_table = (gulong)mono_gc_get_card_table (&card_table_shift, &card_table_mask);
-			gulong nursery_start = (gulong)mono_gc_get_nursery (&nursery_shift, &nursery_size);
+			gulong card_table = (gsize)mono_gc_get_card_table (&card_table_shift, &card_table_mask);
+			gulong nursery_start = (gsize)mono_gc_get_nursery (&nursery_shift, &nursery_size);
```

Comment:
It's probably better to use`GPOINTER_TO_UINT()`

Reply:
I would not want to use those, and demonstrate the questionable pattern of truncation.
It would be correct here, but incorrect to copy to 64bit targets.

Perhaps the macro should use `gsize`, and live with the integer truncation warnings that are already rampant.